pkg/labels/labels.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2019 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package labels
+
+import (
+	"crypto/sha256"
+	"fmt"
+)
+
+// maxLabelLength stores the max label length value supported by kubernetes.
+const maxLabelLength = 63
+
+// GetValidName converts an input string to valid kubernetes label string in accordance to rfc1035 DNS Label spec
+// (https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/identifiers.md)
+func GetValidName(label string) string {
+	// Handle label length
+	finalLabel := handleLabelLength(label)
+
+	return finalLabel
+}
+
+// handleLabelLength adjusts the label length when it exceeds the maxLabelLength (i.e. 63 characters)
+// if length exceeds, we trim the label name to contain only max allowed characters
+// additionally, the last 6 characters of the label name are replaced by the first 6 characters of the sha256 of original label
+func handleLabelLength(label string) string {
+	if len(label) > maxLabelLength {
```

Comment:
I would rather have this code folded under the `GetValidName` function. There is no value in having two functions. It's ok to add the comment from this one to the other one, even if it's gonna be long.

Reply:
Sure. I had created a function as earlier I had yet another function where I was validating the characters for the label (should be alphanumeric etc). Later, I removed that validation as Steve suggested it was not required. I'll move this to GetValidName function.